batteries/dq_dv.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jan  5 10:10:36 2023
+
+@author: mpopeil
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import sklearn as sk
+import sklearn.datasets
+from pandas import DataFrame
+from scipy.interpolate import Akima1DInterpolator
+import scipy
+
+from scipy import signal
+from scipy.signal import savgol_filter
+from scipy import interpolate
+from scipy.integrate import simpson
+from numpy import trapz
+
+#load data into dataframes
+files=['L2_Ta.csv']
+for data in files:
+    
+    df = pd.read_csv(data)
+# charge1=df[(df['Cycle']==1) & (df['Step']==4)]
+# discharge1=df[(df['Cycle']==1) & (df['Step']==7)]
+# charge2=df[(df['Cycle']==2) & (df['Step']==4)]
+# discharge2=df[(df['Cycle']==2) & (df['Step']==7)]
+# charge3=df[(df['Cycle']==3) & (df['Step']==4)]
+# discharge3=df[(df['Cycle']==3) & (df['Step']==7)]
+
+# cycles=[charge1,discharge1, charge2, discharge2, charge3, discharge3]
+
+    for i in range(3,4):
+        charge=df[(df['Cycle']==i) & (df['Step']==4)]
+        discharge=df[(df['Cycle']==i) & (df['Step']==7)]
+        # plt.figure(1)
+        # plt.scatter(charge['Voltage'], charge['Capacity (mAHr)'])
+        # plt.scatter(discharge['Voltage'], discharge['Capacity (mAHr)'])
+        #"clean" data
+        step=(max(charge['Voltage'])-min(charge['Voltage']))/len(charge['Voltage'])
+        x=np.arange(min(charge['Voltage']),max(charge['Voltage']),step)
+        xp=charge['Voltage']
+        xd=discharge['Voltage']
+        fp=charge['Capacity (mAHr)']
+        interpolation=np.interp(x, xp, fp)
+        #smooth data
+        filtered=savgol_filter(interpolation, 9, 1)
```

Comment:
If you do change the scipy import above, this becomes:
```suggestion
        filtered = sp.savgol_filter(interpolation, 9, 1)
```
Note also the suggested spacing around the equals sign, for readability.